### PR TITLE
Feat/276 Table

### PR DESCRIPTION
**Pull request by ShynRou**
Resolves https://taiga.sovryn.app/project/sovryn-light/task/276

Design and Implementation was agreed on with tzu and approved by gilad.

Please review new table on /storybook.

### CODE REVIEW

**Comment by ShynRou** [comment]
Still waiting on input from @t-z-u.

File: src/app/components/Table/index.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  '3xl',
+  'any',
+];
+
+type RowObject = { [param: string]: any };
+
+type ColumnOptions<RowType extends RowObject> = {
+  id: keyof RowType | string;
+  title?: ReactNode;
+  align?: 'left' | 'center' | 'right';
```
Comment (tiltom) [inline src/app/components/Table/index.tsx:25]: Create an enum for this (e.g. `CellAlignment`) and use it instead.

File: src/app/components/Table/index.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  className?: string;
+  hideBelow?: null | Breakpoint;
+  cellRenderer?: (
+    row: RowType,
+    columnId: ColumnOptions<RowType>['id'],
+  ) => ReactNode;
+};
+
+type ITableProps<RowType extends RowObject> = {
```
Comment (tiltom) [inline src/app/components/Table/index.tsx:34]:
```suggestion
type TableProps<RowType extends RowObject> = {
```

File: src/app/components/Table/index.tsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  rows,
+  rowKey,
+  noData,
+  showDetails,
+  detailsTitle,
+  detailsModal,
+}: ITableProps<RowType>) => {
+  const { t } = useTranslation();
+  const [openRow, setOpenRow] = useState<RowType>();
+  const onShowDetails = useCallback(row => showDetails && setOpenRow(row), [
```
Comment (tiltom) [inline src/app/components/Table/index.tsx:58]: Use `if`.

File: src/app/components/Table/index.tsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
+import classNames from 'classnames';
+import React, { ReactNode, useCallback, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { translations } from '../../../locales/i18n';
+import detailsIcon from 'assets/images/ellipsis-h.svg';
+import styles from './index.module.scss';
+import { Dialog } from '../../containers/Dialog';
+
+type Breakpoint = 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | 'any';
```
Comment (tiltom) [inline src/app/components/Table/index.tsx:9]: Convert it to an enum pls.

File: src/app/components/Table/index.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+          </Dialog>
+        ))}
+    </div>
+  );
+};
+
+type ITableRowProps<RowType extends RowObject> = {
+  columns: ITableProps<RowType>['columns'];
+  row: RowType;
+  index: number;
```
Comment (tiltom) [inline src/app/components/Table/index.tsx:184]: Unused.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/components/Table/index.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
+import classNames from 'classnames';
+import React, { ReactNode, useCallback, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { translations } from '../../../locales/i18n';
+import styles from './index.module.scss';
```
Comment (soulBit) [inline src/app/components/Table/index.tsx:5]:
```suggestion
import styles from './index.module.scss';
import rowStyles from './components/TableRow/index.module.scss';
```

File: src/app/components/Table/index.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+                  key={rowKey?.(row) || JSON.stringify(row)}
+                  columns={columns}
+                  row={row}
+                  index={index}
+                  showDetailsBelow={showDetailsBelow}
+                  onShowDetails={onShowDetails}
+                />
+              ))
+            ) : (
+              <tr className={styles.row}>
```
Comment (soulBit) [inline src/app/components/Table/index.tsx:118]:
```suggestion
              <tr className={rowStyles.row}>
```

**Review by soulBit — CHANGES_REQUESTED** [review]
See requested changes that import row styling for cases when no data is present

**Pull request merged**